src/psyclone/psyir/transformations/block_loop_trans.py

```diff
@@ -0,0 +1,258 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2021, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors A. B. G. Chalk STFC Daresbury Lab
+# -----------------------------------------------------------------------------
+
+'''This module provides the BlockLoopTrans, which transforms a Loop into a
+blocked implementation of the Loop'''
+
+from psyclone.core import VariablesAccessInfo, Signature, AccessType
+from psyclone.psyir import nodes
+from psyclone.psyir.nodes import Assignment, BinaryOperation, Reference, \
+        Literal, Loop, Schedule
+from psyclone.psyir.symbols import DataSymbol, ScalarType
+from psyclone.psyir.transformations.loop_trans import LoopTrans
+from psyclone.psyir.transformations.transformation_error import \
+        TransformationError
+
+
+class BlockLoopTrans(LoopTrans):
+    '''
+    Apply a blocking transformation to a loop (in order to permit a
+    chunked parallelisation or improve cache utilisation). For example:
+
```

Comment:
Aha, its that the ".. code-block:: fortran" must be preceded by a blank line otherwise it doesn't parse the text inside as a "code-block". When adding the bank line all warnings disappear.

Reply:
Ok, I've fixed that now too